### ISSUE TITLE
Offset for probalistic modeling - question

### ISSUE THREAD

**Issue by gugatr0n1c** [Evovest/EvoTrees.jl#320]
Hello there,

please, I have question about offset.

I am modeling probability of some event, then I need to run additional model on the top of existing one. Offset is great way how to do it, but I was not able to find how to do it properly and my examples behave really strange.

I am using EvoTreeRegressor, with log_loss loss, target is binary. From first stage model I have probabilities.

Please how do it properly? My current logic is:

a) As offset I am using probabilities (OR should I transform it to logits before?)
b) Should I expect as prediction cummulative value (offset plus new_pred)? Or just new_pred and I need to manually sum offset and new_pred?
c) I would expect cumulative value, but then I need to call predict function with offset as well (for unseen data)? Seems to no way to call predict with offset.
d) Or prediction is in form of logit additional? So need do calc logits from offset, calc sum with new logit pred, and then transform to probabilities?

Actually there is several way how EvoTrees can do it internally. This case with logits is not covered in docs, any info would be great.

Best regards,
Lubomir

**Comment by gugatr0n1c** [comment]
Ok, I ran several runs to figure it out. And seems to me:

a) using probs (not logits) as offset is correct
b) model predict probabilities but from new addiotional logits only
c) for final prediction I need: sigmoid(logit(offset) + logit(new_pred)) = correct probability
d) it is not very intuitive, but in some sence this makes sence

Can you pls confirm this?

e) It would be better that predict function is forced to has offset input when model was built with offset, otherwise it is quite tricky to know what transformation is needed for which case (logit, log, poisson,...)
f) maybe at least for future users mention this in docs

Best regards,
Lubo

**Comment by jeremiedb** [comment]
A offset usage tutorial was added in the docs: https://evovest.github.io/EvoTrees.jl/dev/tutorials/offset-usage

In the future;  I'd agree with you that including the offset effect within the prediction may be most natural and integrate well with the convention where features, target, weight and offset variable names are specified in the `fit` method (though these may eventually move into the model constructor itself, like `EvoTreeRegressor(; target_name = "y", ...)`
